packages/cdktf-cli/lib/project-execution.ts

```diff
@@ -0,0 +1,448 @@
+              }),
+              cond: {
+                type: "onTargetAction",
+                name: "actionIsDiff",
+                value: "diff",
+              },
+            },
+
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "gatherOutput",
+              cond: "planNeedsNoApply",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "approved",
+              cond: "autoApprove",
+            },
+            {
+              actions: assign({
+                targetStackPlan: (_context, event) => event.data,
+              }),
+              target: "waitingForApproval",
+            },
+          ],
+        },
+      },
+      waitingForApproval: {
+        on: {
+          APPROVAL_ABORTED: {
+            target: "done",
+          },
+          APPROVAL_GIVEN: {
+            target: "approved",
+          },
+        },
+      },
+      approved: {
+        always: [
+          {
+            target: "deploy",
+            cond: {
+              type: "onTargetAction",
+              name: "actionIsDiff",
+              value: "deploy",
+            },
+          },
+          {
+            target: "destroy",
+            cond: {
+              type: "onTargetAction",
+              name: "actionIsDiff",
+              value: "destroy",
+            },
+          },
+        ],
+      },
+      deploy: {
+        invoke: {
+          id: "deploy",
+          src: "deploy",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: {
+            target: "gatherOutput",
+          },
+        },
+      },
+      destroy: {
+        invoke: {
+          id: "destroy",
+          src: "destroy",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: {
+            target: "gatherOutput",
+          },
+        },
+      },
+      gatherOutput: {
+        invoke: {
+          id: "gatherOutput",
+          src: "gatherOutput",
+          onError: {
+            target: "error",
+            actions: assign({
+              message: (_context, event) => event.data,
+            }),
+          },
+          onDone: {
+            target: "done",
+            actions: assign({
+              outputs: (_context, event) => event.data.outputs,
+              outputsByConstructId: (_context, event) =>
+                event.data.outputsByConstructId,
+            }),
+          },
+        },
+      },
+      done: {
+        type: "final",
+      },
+      error: {
+        type: "final",
+      },
+    },
+  },
+  { services, guards: guards as any }
```

Comment:
Is `as any` really necessary? Can't this be typed?

Reply:
yes, the xstate types are lacking, they can't handle custom guards right now